### PR TITLE
Simplify/rewrite

### PR DESCRIPTION
**Pull request by kesla**
Send in html, get amp out.

R = @iefserge 
T = major

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import setupHtmlToArticleJson from 'html-to-article-json';
+import articleJsonToAmp from 'article-json-to-amp';
+import Promise from 'pinkie-promise';
```
Comment (iefserge) [inline lib/index.js]: just curious why not Bluebird?

Comment (kesla) [inline lib/index.js]: @iefserge pinkie-promise return the native promise if available. I saw someone using it so I figured it'd make sense. I'm fine using Bluebird if you have experience with that and you know it's good!

Comment (iefserge) [inline lib/index.js]: @kesla https://www.npmjs.com/package/native-or-bluebird

File: lib/index.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import imageSize from 'request-image-size';
+
+const addDimensions = (json) => Promise.all(
+  json.map(row => {
+    if (row.embedType !== 'image' || (row.width && row.height) || !row.src) {
+      return row;
+    }
+
+    return new Promise((resolve, reject) => {
+      imageSize(row.src, (err, dimensions) => {
```
Comment (iefserge) [inline lib/index.js]: timeout needed

File: lib/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    });
+  })
+);
+
+module.exports = () => {
+  const htmlToArticleJson = setupHtmlToArticleJson();
+
+  const htmlToAmp = html => Promise.resolve(htmlToArticleJson(html))
+    .then(json => addDimensions(json))
+    .then(json => Promise.resolve(articleJsonToAmp(json)));
```
Comment (iefserge) [inline lib/index.js:34]: nice!

File: lib/index.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  return (html, cb) => {
+    if (cb) {
+      htmlToAmp(html).then(amp => cb(null, amp), err => cb(err));
+    } else {
+      return htmlToAmp(html);
```
Comment (iefserge) [inline lib/index.js:40]: why return here?

File: readme.md [hunk 5]
```diff
@@ -21,17 +39,18 @@ npm test
...
 ## Dev Dependencies
 
+- [ava](https://github.com/sindresorhus/ava): Futuristic test runner 🚀
+- [babel-cli](https://github.com/babel/babel/tree/master/packages): Babel command line.
+- [babel-core](https://github.com/babel/babel/tree/master/packages): Babel compiler core.
+- [babel-preset-es2015](https://github.com/babel/babel/tree/master/packages): Babel preset for all es2015 plugins.
+- [nyc](https://github.com/bcoe/nyc): a code coverage tool that works well with subprocesses.
+- [package-json-to-readme](https://github.com/zeke/package-json-to-readme): Generate a README.md from package.json contents
 - [semistandard](https://github.com/Flet/semistandard): All the goodness of `feross/standard` with semicolons sprinkled on top.
 - [snazzy](https://github.com/feross/snazzy): Format JavaScript Standard Style as Stylish (i.e. snazzy) output
```
Comment (iefserge) [inline readme.md:67]: not sure how useful dependencies sections are in the readme

File: readme.md [hunk 6]
```diff
@@ -10,6 +10,24 @@ Download node at [nodejs.org](http://nodejs.org) and install it, if you haven't
 npm install html-to-amp --save
 ```
 
+## Usage
```
Comment (iefserge) [inline readme.md:13]: I'd add more info about the image requirements, supported embeds etc

**Comment by iefserge** [comment]
@kesla looks very good, some minor comments, other than that LGTM :+1:

File: readme.md (hunk 5 above)
Comment (kesla) [inline readme.md:67]: they're automatically generated by https://www.npmjs.com/package/package-json-to-readme

File: lib/index.js (hunk 4 above)
Comment (kesla) [inline lib/index.js:40]: returns the promise

Comment (iefserge) [inline lib/index.js:40]: got it, but other branch returns nothing?

Comment (kesla) [inline lib/index.js:40]: Yes. It runs the callback

Comment (iefserge) [inline lib/index.js:40]: ok

**Comment by kesla** [comment]
@iefserge thanks for really good feedback, updated usage section & fixed some other of your comments. So this is now ready to go. Boom!

**Comment by iefserge** [comment]
:+1:

**Pull request merged**